src/snowcli/cli/common/decorators.py

```diff
@@ -101,10 +150,16 @@ def _extend_signature_with_global_options(func):
         )
         for p in existing_parameters
     ]
-    sig = sig.replace(
-        parameters=[
+    if with_connection:
+        parameters = [
             *existing_parameters_with_evaluated_types,
             *GLOBAL_CONNECTION_OPTIONS,
+            *GLOBAL_OPTIONS,
+        ]
+    else:
+        parameters = [
+            *existing_parameters_with_evaluated_types,
+            *GLOBAL_OPTIONS,
         ]
-    )
+    sig = sig.replace(parameters=parameters)
```

Comment:
Instead of this if I would be in favor of passing array of prameters that should be added. WDYT?

Reply:
It looks much better 👍